File: 2265-partition-array-according-to-given-pivot/2265-partition-array-according-to-given-pivot.py
class Solution:
    def pivotArray(self, nums, pivot):
        less = 0
        equal = 0
        for num in nums:
            if num < pivot:
                less += 1
            elif num == pivot:
                equal += 1

        ans = [0] * len(nums)
        lessI = 0
        equalI = less
        greaterI = less + equal
        for i in range(len(nums)):
            num = nums[i]
            if num < pivot:
                ans[lessI] = num
                lessI += 1
            elif num > pivot:
                ans[greaterI] = num
                greaterI += 1
            else:
                ans[equalI] = num
                equalI += 1

        return ans

File: 2265-partition-array-according-to-given-pivot/2265-partition-array-according-to-given-pivot.rb
# @param {Integer[]} nums
# @param {Integer} pivot
# @return {Integer[]}


def pivot_array(nums, pivot)
    ans = []
    n = nums.length 

    equal_count = 0 
    bigger = []
    
    for num in nums do 
        if num == pivot
            equal_count += 1
        elsif num < pivot
            ans << num
        else
            bigger << num
        end 
    end 

    equal_count.times do 
        ans << pivot 
    end 

    bigger.each do |num|
        ans << num
    end 

    return ans 

     

end